app/models/vacation.rb

```diff
@@ -2,13 +2,53 @@ class Vacation < ApplicationRecord
   belongs_to :employee
 
   validates :started_on, presence: true
-  validates :finished_on, presence: true
-  validates :employee_id, presence: true, uniqueness: true
-  validate :start_and_finish_dates
-end
+  validates :finished_on, presence: true, comparison: { greater_than: :started_on }
+  validates :employee_id, presence: true
+  validate :remaining_vacation_days
+  validate :not_past_date
+  validate :vacation_overlap
+  validate :position_active
+  validate :can_take_vacation
+
+  def remaining_vacation_days
+    remaining_days = employee.vacation_days - employee.used_vacation_days
+    return if remaining_days > 0
+    errors.add(:vacation_days, "Dont have enough days")
+  end
+
+  def not_past_date
+    return unless started_on && finished_on
+    return if started_on > DateTime.now
+    errors.add(:vacation_days, "You cannot take vacation in a past date")
+  end
 
-def start_and_finish_dates
-  if  started_on && finished_on && started_on > finished_on
-    errors.add(:finished_on, "Vacation cannot end before it begins")
+  def vacation_overlap
+    employee.vacations.each do |vacation|
+      vacation_interval = vacation.started_on..vacation.finished_on
+      if vacation_interval.cover?(started_on) || vacation_interval.cover?(finished_on)
+        errors.add(:vacation_days, "Vacation cannot overlap with other vacations")
+      end
+    end
   end
+
+  def position_active
+    last_position_history = employee.position_histories.last
+    if last_position_history.present? && last_position_history.finished_on.present?
+      errors.add(:vacation_days, "You can't add vacation because the last position is already finished.")
+    end
+  end
+
+  def can_take_vacation
+    department = employee.department
```

Comment:
For now, it makes the next number of queries into db:
1. To get a department. (1 query) 
2. To get all employees of the department. (1 query)
3. To get all vacations of all employees. ( 1 query per employee)

And now, imagine, how many queries it makes in order to calculate this. Each query costs money on production, so let's save money for us or our customers.
You have to check if the employee can take a vacation using  **_only_** 1 queries and no more. As a hint, it should starts with:
`Vacation.joins(:employee).where...`